code/controllers/subsystem/weeds_decay.dm

```diff
@@ -0,0 +1,51 @@
+SUBSYSTEM_DEF(weeds_decay)
+	name = "Weed Decay"
+	priority = FIRE_PRIORITY_WEED
+	runlevels = RUNLEVEL_LOBBY|RUNLEVEL_SETUP|RUNLEVEL_GAME|RUNLEVEL_POSTGAME
+	wait = 15 SECONDS
+
+	// This is a list of nodes on the map.
+	var/list/decaying = list()
+
+/datum/controller/subsystem/weeds_decay/stat_entry()
+	return ..("Decay Nodes: [length(decaying)]")
+
+/datum/controller/subsystem/weeds_decay/fire(resumed = FALSE)
+
+	for(var/A in decaying)
+		if(MC_TICK_CHECK)
+			return
+
+		var/turf/T = A
+		if(QDELETED(T))
+			decaying -= T
+			continue
+
+		var/obj/effect/alien/weeds/W = locate() in T
+		if(QDELETED(W) || W.parent_node)
+			decaying -= T
+			continue
+
+		var/decay_chance = 100
+		for(var/direction in GLOB.cardinals) 
+			var/turf/adj = get_step(T, direction)
+			if(locate(/obj/effect/alien/weeds) in adj)
+				decay_chance -= rand(25, 40)
```

Comment:
`decay_chance` is already a probability value. Don't stack probability over probability. There's 9 possible adjacent weeds, so -10% each will still keep a low 10% chance of happening on a healthy but nodeless network.

Reply:
Im only checking cardinals because it made it a little better for the check (and was also how it grows), so instead of 10% ill do 25% each time